pkg/crdexplorer/defaultexplorer/default.go

```diff
@@ -33,12 +35,12 @@ func (e *DefaultExplorer) HookEnabled(kind schema.GroupVersionKind, operationTyp
 			return true
 		}
 	case configv1alpha1.ExploreRetaining:
-		if _, exist := e.retentionHandlers[kind]; exist {
-			return true
-		}
-
-		// TODO(RainbowMango): more cases should be added here
+		// for retaining explorer, always enabled for all GVK.
+		return true
```

Comment:
I DO NOT THINK SO. Why always enabled??

Reply:
former functions do not return error when kind is not expected., and just return nil.
https://github.com/karmada-io/karmada/blob/02c72b520614958c065bfb2e04d24aef903293dc/pkg/util/objectwatcher/retain.go#L39-L52

if not doing like this e2e will fail, and I found `deployment` and `endpointslices` will also trigger retain functions. and maybe there are some other resource kinds will trigger retain explorer, so for simplify, I enable all retain explorers. keep compatible with old functions.

Reply:
error logs may be like this
Failed to retain fields for resource(kind=Deployment, karmadatest-dt4/deploy-kh2) 
in cluster member1: default retain explorer for Deployment not found